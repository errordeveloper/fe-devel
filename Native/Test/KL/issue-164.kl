struct MyType {
  String string;
};

// Construct from a string
function MyType(String string) {
  this.string = "The string was '" + string + "'";
}

// Construct from a scalar
function MyType(Scalar scalar) {
  this.string = "The scalar was " + scalar;
}

operator entry() {
  MyType myTypeFromString = "foo";
  report(myTypeFromString);
  MyType myTypeFromScalar = 2.718;
  report(myTypeFromScalar);
}
