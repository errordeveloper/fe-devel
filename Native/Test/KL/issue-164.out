{string:"The string was 'foo'"}
{string:"The scalar was 2.718"}
